cips/cip-X.md

```diff
@@ -0,0 +1,271 @@
+      {
+        "type": "offchain/MsgSignArbitraryData",
+        "value": {
+          "appDomain": "simd",
+          "signerAddress": "cosmos1hftz5ugqmpg9243xeegsqqav62f8hnywsjr4xr",
+          "data": "cmFuZG9t"
+        }
+      }
+    ],
+    "fee": {
+      "amount": [],
+      "gas": "0"
+    },
+    "signatures": [
+      {
+        "pub_key": {
+          "type": "tendermint/PubKeySecp256k1",
+          "value": "AqnDSiRoFmTPfq97xxEb2VkQ/Hm28cPsqsZm9jEVsYK9"
+        },
+        "signature": "8y8i34qJakkjse9pOD2De+dnlc4KvFgh0wQpes4eydN66D9kv7cmCEouRrkka9tlW9cAkIL52ErB+6ye7X5aEg=="
+      }
+    ],
+    "memo": ""
+  }
+}
+```
+
+### MsgSignIn
+
+The second message added to the `offchain` package is `MsgSignIn`.
+
+`MsgSignIn` enables the proof of wallet ownership for applications sign-in. In this context, `AppDomain` is the
+application requesting off-chain message signing. `URI` refers to the resource that is the subject of the signing.
+`signerAddress` is the account address of the signer. `Nonce` is a random string typically chosen by the relying on party and
+used to prevent replay attacks. `Issued-at` the time when the message was generated.
+
+Proto definition:
+
+```protobuf
+// MsgSignArbitraryData defines an arbitrary, general-purpose, off-chain message
+message MsgSignIn {
+  option (cosmos.msg.v1.signer) = "signerAddress";
+  // AppDomain is the application requesting off-chain message signing
+  string appDomain = 1;
+  // Uri is the resource that is the subject of the signing
+  string uri = 2;
+  // Signer is the sdk.AccAddress of the message signer
+  string signerAddress = 3 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // Nonce is a random string
+  string nonce = 4;
+  // Issued-at is the time when the message was generated
+  google.protobuf.Timestamp issued-at = 5;
+}
+```
+
+Signed MsgSignIn json example:
+
+```json
+{
+  "type": "cosmos-sdk/StdTx",
+  "value": {
+    "msg": [
+      {
+        "type": "offchain/MsgSignIn",
+        "value": {
+          "appDomain": "exampleSwap",
+          "uri": "https://exampleSwap.com/login",
+          "signerAddress": "cosmos1hftz5ugqmpg9243xeegsqqav62f8hnywsjr4xr",
+          "nonce": "14368412",
+          "issued-at": "2023-10-10T16:25:24Z"
+        }
+      }
+    ],
+    "fee": {
+      "amount": [],
+      "gas": "0"
+    },
+    "signatures": [
+      {
+        "pub_key": {
+          "type": "tendermint/PubKeySecp256k1",
+          "value": "AqnDSiRoFmTPfq97xxEb2VkQ/Hm28cPsqsZm9jEVsYK9"
+        },
+        "signature": "8y8i34qJakkjse9pOD2De+dnlc4KvFgh0wQpes4eydN66D9kv7cmCEouRrkka9tlW9cAkIL52ErB+6ye7X5aEg=="
+      }
+    ],
+    "memo": ""
+  }
+}
+```
+
+## Drawbacks
+
+This CIP requires a fixed relationship between an account address and a public key. That means it won't work if [ADR-034](https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-rc.1/docs/architecture/adr-034-account-rekeying.md) 
+is implemented.
+
+Doesn't work with multisig accounts.
```

Comment:
is this true? aren't all multi-sig pubkeys posted in a deterministic fashion now so the mapping is unique

Reply:
it should work with multisigs as the current multisig is deterministic, but on chain multisigs would not work here. It might make sense to clarify this.

Reply:
gentle ping on adding some verbiage here